src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs

```diff
@@ -171,15 +181,103 @@ public override async Task ExecuteAsync(InstanceDetailsViewModel model)
                 }
             }
 
+            if (!instance.VersionHasServiceControlAuditFeatures)
```

Comment:
What about the case where audit was already disabled?

Reply:
That's in the PoA to be done ;) We can include the check it in this PR though.